File: public/algorithms/13.RomantoInteger/13.RomantoInteger.cpp
int romanToInt(string s) {
    unordered_map<char, int> m = {
        {'I',1},{'V',5},{'X',10},{'L',50},
        {'C',100},{'D',500},{'M',1000}
    };
    int total = 0;
    for (int i = 0; i < s.size(); ++i) {
        if (i + 1 < s.size() && m[s[i]] < m[s[i + 1]]) {
            total -= m[s[i]];
        } else {
            total += m[s[i]];
        }
    }
    return total;
}

File: public/algorithms/13.RomantoInteger/13.RomantoInteger.js
var romanToInt = function (s) {
  const map = {
    I: 1,
    V: 5,
    X: 10,
    L: 50,
    C: 100,
    D: 500,
    M: 1000,
  };
  let total = 0;
  for (let i = 0; i < s.length; i++) {
    let curr = map[s[i]];
    let next = map[s[i + 1]] || 0;
    if (curr < next) total -= curr;
    else total += curr;
  }
  return total;
};
